src/inference/src/os/lin/lin_system_conf.cpp

```diff
@@ -43,13 +43,85 @@ CPU::CPU() {
         return 0;
     };
 
+    auto CheckValidCpu = [&]() {
+        cpu_set_t mask;
+        CPU_ZERO(&mask);
+
+        if (sched_getaffinity(0, sizeof(cpu_set_t), &mask) == -1) {
+            return -1;
+        }
+
+        int total_proc = 0;
+        std::vector<int> cores_list;
+        std::vector<int> phy_core_list;
+        std::vector<std::vector<int>> valid_cpu_mapping_table;
+
+        for (int i = 0; i < _processors; i++) {
+            if (CPU_ISSET(i, &mask)) {
+                total_proc++;
+                cores_list.emplace_back(_cpu_mapping_table[i][CPU_MAP_CORE_ID]);
+                valid_cpu_mapping_table.emplace_back(_cpu_mapping_table[i]);
+                if (_cpu_mapping_table[i][CPU_MAP_CORE_TYPE] == MAIN_CORE_PROC) {
+                    phy_core_list.emplace_back(_cpu_mapping_table[i][CPU_MAP_GROUP_ID]);
+                }
+            }
+        }
+
+        if (total_proc == _processors) {
```

Comment:
Is it considered the possible of `total_proc==0` ?

Reply:
updated